op-chain-ops/cmd/check-jovian/main.go

```diff
@@ -0,0 +1,291 @@
+	if err != nil {
+		return fmt.Errorf("failed to create bindings around L1Block contract: %w", err)
+	}
+	daFootprintGasScalar, err := cl.DaFootprintGasScalar(nil)
+	if err != nil {
+		return fmt.Errorf("failed to get DA footprint gas scalar from L1Block contract: %w", err)
+	}
+	if daFootprintGasScalar == 0 {
+		env.log.Warn("DA footprint gas scalar is set to 0. SystemConfig needs to emit scalar change to update.")
+	}
+	env.log.Info("L1Block test: success", "daFootprintGasScalar", daFootprintGasScalar)
+	return nil
+}
+
+// checkBlock checks that the latest block header has a non-nil blobgasused field
+// If a secret key is provided, it will attempt to send a tx-to-self on L2, wait for it to be mined,
+// then use the block containing that tx as the block to check.
+func checkBlock(ctx context.Context, env *actionEnv) error {
+	var err error
+
+	var latest *types.Block
+
+	// If a secret key was provided, attempt to send a tx-to-self and wait for it to be mined.
+	if env.secretKey != "" {
+		env.log.Info("secret key provided - attempting to send tx-to-self and wait for inclusion")
+
+		cfg := txmgr.NewCLIConfig(env.l2endpoint, txmgr.DefaultBatcherFlagValues)
+		cfg.PrivateKey = env.secretKey
+		t, err := txmgr.NewSimpleTxManager("check-jovian", env.log, new(metrics.NoopTxMetrics), cfg)
+		fromAddr := t.From()
+
+		if err != nil {
+			return fmt.Errorf("failed to create tx manager: %w", err)
+		}
+		defer t.Close()
+
+		receipt, err := t.Send(ctx, txmgr.TxCandidate{
+			To:    &fromAddr, // Send to self
+			Value: big.NewInt(0),
+		})
+		if err != nil {
+			return fmt.Errorf("error waiting for tx to be mined: %w", err)
+		}
+		if receipt == nil {
+			return fmt.Errorf("tx mined receipt was nil")
+		}
+
+		env.log.Info("tx mined", "txHash", receipt.TxHash.Hex(), "blockNumber", receipt.BlockNumber.Uint64(), "blobGasUsed", receipt.BlobGasUsed)
+
+		if receipt.BlobGasUsed == 0 {
+			return fmt.Errorf("receipt.BlobGasUsed was zero (required with Jovian)")
+		}
+
+		// Fetch the block that contained the receipt
+		blk, err := env.l2.BlockByNumber(ctx, receipt.BlockNumber)
+		if err != nil {
+			return fmt.Errorf("failed to fetch block containing tx: %w", err)
+		}
+		latest = blk
+	} else {
+		latest, err = env.l2.BlockByNumber(ctx, nil)
+		if err != nil {
+			return fmt.Errorf("failed to get latest block: %w", err)
+		}
+	}
+
+	bgu := latest.BlobGasUsed()
+	if bgu == nil {
+		return fmt.Errorf("block %d has nil BlobGasUsed field", latest.Number())
+	}
+
+	txs := latest.Body().Transactions
+
+	// A non-zero BlobGasUsed is hard evidence of Jovian being active
+	// A zero value is inconclusive (could be an empty block or pre-Jovian)
+	if *bgu == 0 {
+		env.log.Warn("Block header BlobGasUsed is zero - inconclusive for Jovian activation",
+			"blockNumber", latest.Number(),
+			"note", "Zero could indicate an empty block or pre-Jovian state")
+
+		if len(txs) > 1 && txs[len(txs)-1].Type() == types.DepositTxType {
```

Comment:
```suggestion
		if len(txs) > 1 && txs[len(txs)-1].Type() != types.DepositTxType {
```
the last tx must _not_ be a deposit tx for it to be a user tx.

But note that you could change the outer branch on line 221 just be `if len(tx) == 1` and then only print the warning log, and let the `else` branch also take care of 0 DA footprint.

Reply:
Good spot, I took your second suggestion.